### DISCUSSION TITLE
[vite-node] set NODE_ENV to production

### DISCUSSION THREAD

**Discussion by pietrovismara** [vitest-dev/vitest#2692] [Q&A] [answered]
I'm trying to set NODE_ENV when running `vite-node`:

```
NODE_ENV=production vite-node --options.mode=production src/index.ts
```

What happens is that `vite-node` overrides NODE_ENV to `development`.

Is it possible (and is it a good idea) to use `vite-node` to run nodejs in a production environment?

Alternatively is there a way to override the `mode` of `vite` when launching `vite-node`?

**Comment by Johann-S** [comment, marked as answer]
I saw here: https://github.com/vitest-dev/vitest/discussions/4420#discussioncomment-7444401

That `node-vite` starts 2 node processes that summed take 6 times more memory than Node alone... So currently I'm not sure if it's a good idea to use it in production. That's what I would like to do too 😅 

Unfortunately I don't know what's the purpose of those two processes.. Because maybe if you disable the `watch` mode, only one process will stay?! 🤔

**Reply by pietrovismara** [reply]
Gave up with vite-node a long time ago but thanks for the insight

**Comment by hyunbinseo** [comment]
You can override the `import.meta.env.MODE` like this:

```shell
pnpm vite-node --mode=production cli/index.ts
```

Note that only the `MODE` value has been updated:

```js
{
  BASE_URL: '/',
  DEV: true,
  MODE: 'production', // updated
  PROD: false, // is still false
  SSR: true
}
```

This seems to be expected according to the [Vite docs](https://vite.dev/guide/env-and-mode#node-env-and-modes).
